controllers/utils/multicloud_util.go

```diff
@@ -0,0 +1,341 @@
+		// Get the InstallPlan content from the ManagedClusterView.
+		installPlan := operatorsv1alpha1.InstallPlan{}
+		json.Unmarshal([]byte(mcvForInstallPlan.Status.Result.Raw), &installPlan)
+
+		// If the InstallPlan's approval is not manual, return. No action is taken for Automatic install plans.
+		if installPlan.Spec.Approval != operatorsv1alpha1.ApprovalManual {
+			Log.Info("InstallPlan can't be approved as it's approval is already set to Automatic",
+				"installPlan", installPlan.ObjectMeta.Name, "namespace", installPlan.ObjectMeta.Namespace)
+			return InstallPlanCannotBeApproved, nil
+		}
+
+		// If the InstallPlan has already been approved, return.
+		if installPlan.Spec.Approval != operatorsv1alpha1.ApprovalManual {
+			Log.Info("InstallPlan has already been approved",
+				"installPlan", installPlan.ObjectMeta.Name, "namespace", installPlan.ObjectMeta.Namespace)
+			return InstallPlanCannotBeApproved, nil
+		}
+
+		Log.Info("[ProcessSubscriptionManagedClusterView] Create ManagedClusterAction for InstallPlan", "InstallPlan",
+			installPlan.ObjectMeta.Name, "namespace", installPlan.ObjectMeta.Namespace)
+		// Create or update the managedClusterAction to approve the install plan.
+		mcaName := GetMultiCloudObjectName(
+			ManagedClusterActionPrefix, clusterGroupUpgrade, "InstallPlan", installPlan.Name)
+		_, err := EnsureManagedClusterActionForInstallPlan(ctx, c, mcaName, clusterName, installPlan, Log)
+		if err != nil {
+			return InstallPlanCannotBeApproved, err
+		}
+
+		// Check that the ManagedClusterAction completed successfully. If so, then there is no need to also check
+		// the install plan.
+		time.Sleep(ManagedClusterActionWaitTimeSec * time.Second)
+		err = CheckManagedClusterActionDone(ctx, c, mcaName, clusterName, Log)
+		if err != nil {
+			Log.Info("There was an issue with the ManagedClusterAction for InstallPlan", "error", err.Error())
+			return InstallPlanCannotBeApproved, err
+		}
+		return InstallPlanWasApproved, nil
+	}
+
+	return InstallPlanCannotBeApproved, nil
+}
+
+// EnsureManagedClusterView creates or updates a view.
+func EnsureManagedClusterView(
+	ctx context.Context, c client.Client, name string, namespace string, resourceType string,
+	resourceName string, resourceNamespace string, cguLabel string, Log logr.Logger) (*viewv1beta1.ManagedClusterView, error) {
+
+	mcv := &viewv1beta1.ManagedClusterView{}
+	err := c.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, mcv)
+
+	if err != nil {
+		// If the specific managedClusterView was not found, create it.
+		if errors.IsNotFound(err) {
+			Log.Info("[EnsureManagedClusterView] MCV doesn't exist, create it", "name", name, "namespace", namespace)
+			viewMeta := metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				Labels: map[string]string{
+					"openshift-cluster-group-upgrades/clusterGroupUpgrade": cguLabel,
```

Comment:
nit: "openshift-cluster-group-upgrades/clusterGroupUpgrade"is being frequently used. Consider defining a const var.

Reply:
I think this is the case with a couple of other labels. Will add it to the enhancements list.